dio/lib/src/utils.dart

```diff
@@ -38,8 +38,8 @@ String encodeMap(
 }) {
   final urlData = StringBuffer('');
   bool first = true;
-  final leftBracket = isQuery ? '[' : '%5B';
-  final rightBracket = isQuery ? ']' : '%5D';
+  final leftBracket = isQuery || !encode ? '[' : '%5B';
```

Comment:
I think we can add some related comments to avoid breaking again.

Reply:
Done